include/xmp.h

```diff
@@ -16,12 +16,11 @@ extern "C" {
 #define XMP_VER_RELEASE 0
 
 #if defined(_WIN32) && !defined(__CYGWIN__)
-# if defined(BUILDING_STATIC)
-#  define LIBXMP_EXPORT
-# elif defined(BUILDING_DLL)
+# if defined(BUILDING_DLL)
 #  define LIBXMP_EXPORT __declspec(dllexport)
 # else
-#  define LIBXMP_EXPORT __declspec(dllimport)
+#  define LIBXMP_EXPORT
+#  define LIBXMP_EXPORT_VAR __declspec(dllimport)
```

Comment:
This patch is wrong: it attaches `dllimport` attribute to vars in static builds
which we don't want at all.

Reply:
You right, I should avoid this line being used when `BUILDING_STATIC` is enabled. I'll fix this tomorrow, now I'm at the bed from the phone.

Reply:
Fixed just now, please verify

Reply:
And now you rely on `BUILDING_STATIC` again which you want to avoid..

Reply:
> And now you rely on BUILDING_STATIC again which you want to avoid..

The big problem was all calls in general, the last thing left were these constants. To deal with that I added calls to let users use them instead of variables, but, variables weren't removed to keep the compatibility with old code that already uses `BUILDING_STATIC` at their side. So, these variables should be just deprecated, not removed (until the next major)

Reply:
Also, the note: presence/absence of the `dllimport` affects only the client-side compilation and absolutely doesn't affects the compatibility with the existing DLL built. I mean, when there are two clients built with and without `dllimport`, they will still work fine.